cmd/thanos/compact.go

```diff
@@ -34,6 +68,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	objStoreConfig := regCommonObjStoreFlags(cmd, "")
 
+	currentMaxLevel := len(compactions) - 1
+	maxCompactionLevel := cmd.Flag("compaction.max-level", fmt.Sprintf("Maximum compaction level, default is %d: %s", currentMaxLevel, compactions.String())).Default(strconv.Itoa(currentMaxLevel)).Int()
```

Comment:
Let's make it Hidden:
```suggestion
	maxCompactionLevel := cmd.Flag("compaction.max-level", fmt.Sprintf("Maximum compaction level, default is %d: %s", currentMaxLevel, compactions.String())).Default(strconv.Itoa(currentMaxLevel)).Hidden().Int()
```
as this is not recommended to change this.

Reply:
👌